src/models/chainspec.rs

```diff
@@ -0,0 +1,470 @@
+use std::collections::HashMap;
+
+use ethereum_types::{H160, H256};
+use serde::Deserialize;
+
+type NodeUrl = String;
+
+#[derive(Debug, Deserialize)]
+struct ChainSpec {
+    name: String,
+    data_dir: String,
+    bootnodes: Vec<NodeUrl>,
+    engine: Engine,
+    hardforks: HardForks,
+    params: Params,
+    genesis: Genesis,
+    precompiles: HashMap<String, HashMap<String, Precompiles>>,
+    balances: HashMap<H160, String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Engine {
+    name: String,
+    params: EngineParams,
+}
+
+#[derive(Debug, Deserialize)]
+struct EngineParams {
+    period: f32,
+    epoch: f32,
+    genesis: EngineGenesis,
+}
+
+#[derive(Debug, Deserialize)]
+struct EngineGenesis {
+    vanity: H256,
+    signers: Vec<H160>,
+}
+
+#[derive(Debug, Deserialize)]
+struct HardForks {
+    #[serde(flatten)]
+    eips: HashMap<String, String>,
+    max_code_size: String,
```

Comment:
Actually, this, and everywhere else, is `u64`. `primitive_types::U64` has the code to parse hex strings.